services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java

```diff
@@ -0,0 +1,61 @@
+package app.coronawarn.server.services.distribution.assembly.component;
+
+import app.coronawarn.server.common.protocols.internal.dgc.ValueSets;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.ArchiveOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.decorator.signing.DistributionArchiveSigningDecorator;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.DirectoryOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.file.FileOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig.DigitalGreenCertificate;
+import app.coronawarn.server.services.distribution.dgc.DigitalGreenCertificateToProtobufMapping;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * Reads configuration parameters from the respective files in the class path and build a {@link
+ * DigitalGreenCertificateStructureProvider} with them.
+ */
+@Component
+public class DigitalGreenCertificateStructureProvider {
+
+  private static final Logger logger = LoggerFactory.getLogger(DigitalGreenCertificateStructureProvider.class);
+
+  private final DistributionServiceConfig distributionServiceConfig;
+  private final CryptoProvider cryptoProvider;
+  private final DigitalGreenCertificateToProtobufMapping dgcToProtobufMapping;
+
+  /**
+   * Create an instance.
+   */
+  public DigitalGreenCertificateStructureProvider(DistributionServiceConfig distributionServiceConfig,
+      CryptoProvider cryptoProvider, DigitalGreenCertificateToProtobufMapping dgcToProtobufMapping) {
+    this.distributionServiceConfig = distributionServiceConfig;
+    this.cryptoProvider = cryptoProvider;
+    this.dgcToProtobufMapping = dgcToProtobufMapping;
+  }
+
+  /**
+   * Returns the publishable archive with the Digital Green Certificates protobuf structure for mobile clients.
+   */
+  public DirectoryOnDisk getDigitalGreenCertificates() {
+    return constructArchiveToPublish(distributionServiceConfig.getDigitalGreenCertificate(),
+        dgcToProtobufMapping.constructProtobufMapping());
+  }
+
+  private DirectoryOnDisk constructArchiveToPublish(
```

Comment:
needs to be enhanced with further development by additional languages

Reply:
Being implemented right now and is documented by EXPOSUREBACK-1270